Texstyle/Texstyle/UIButton+Text.swift

```diff
@@ -6,10 +6,44 @@ import UIKit
 
 extension UIButton {
 
-    /// Applies the text for all states.
-    public func setText(_ text: Text) {
-        for state in text.styles.keys {
+    /// Applies the text for passed style
+    ///
+    /// - Parameters:
+    ///   - text: A text that should be applied
+    ///   - state: Specific control state for this text
+    public func setText(_ text: Text, for state: ControlState = .normal) {
+        setAttributedTitle(text.attributed, for: state)
+    }
+
+    /// Applies the text for passed style
+    ///
+    /// - Parameters:
+    ///   - text: A text that should be applied
+    public func setText(_ text: ControlStateText) {
+        let states = Array(Set(Array(text.styles.keys) + Array(text.substyles.keys)))
```

Comment:
Looks... interesting. As I understood, ControlStateText may have only normal style and only selected substyle. According to `attributed` function implementation, string for selected state will be nil. In this case you can enumerate only `text.styles.keys`